site/docs/sql/00-intro.md

```diff
@@ -0,0 +1,202 @@
+---
+title: Hazelcast Jet SQL
+description: Introduction to Hazelcast Jet SQL features.
+---
+
+Hazelcast Jet allows you to create a Jet processing job using the
+familiar SQL language. It can execute distributed SQL statements over
+Hazelcast IMaps and external data sets.
+
+**Note:** _The service is in beta state. Behavior and API might change
+in future releases. Binary compatibility is not guaranteed between minor
+or patch releases._
+
+## Overview
+
+In the first release, Jet SQL supports the following features:
+
+- SQL Queries over [Apache Kafka topics](05-kafka-connector.md) and
+[files (local and remote)](04-files-connector.md)
+- Joining Kafka or file data with local IMaps (enrichment)
+- Filtering and projection using [SQL
+expressions](https://docs.hazelcast.org/docs/{imdg-version}/manual/html-single/index.html#expressions)
+- Aggregating data from files using predefined
+[aggregate functions](00a-basic-commands#aggregation-functions)
+- Receiving query results via Jet client (Java) or writing the results
+to an [IMap](03-imap-connector.md) in the Jet cluster
+- Running continuous (streaming) and batch queries, see
+[Job Management](02-job-management.md)
+
+These are some of the features on our roadmap:
+
+- Joins with arbitrary external data sources
+- Windowed aggregation
+- JDBC
+
+## Example: How to query Apache Kafka using SQL
+
+We have a topic called `trades` with stock trade events. Trades are
+encoded as JSON messages:
+
+```json
+{ "ticker": "ABCD", "price": 5.5 }
+```
+
+To use a remote topic as a table in Jet, first create an `EXTERNAL
+MAPPING`. It maps the JSON messages to a fixed list of columns with
+data types:
+
+```sql
+CREATE MAPPING trades (
+    ticker VARCHAR,
+    value DECIMAL)
+TYPE Kafka
+OPTIONS (
+    valueFormat 'json',
+    "bootstrap.servers" '1.2.3.4',
+    "value.deserializer" 'org.apache.kafka.connect.json.JsonSerializer')
+```
+
+To submit the above query, use the Java API (we plan JDBC support and
+support in non-Java clients in the future):
+
+```java
+JetInstance inst = ...;
+inst.getSql().execute( /* query text */ );
+```
+
+A SQL query can now be used to read from the `trades` topic, as if it
+was a table:
+
+```java
+JetInstance inst = ...;
+try (SqlResult result = inst.getSql().execute("SELECT * FROM trades")) {
+    for (SqlRow row : result) {
```

Comment:
Since `trades`  is streaming job which means it will never finish isn't this actually example of "how this should not be used"?

Reply:
Why not? You can have a command-line client that prints the contents of a topic, until it's killed with ctrl-c. It's clearly noted below.